src/platform/datapath_winuser.c

```diff
@@ -3021,73 +3042,72 @@ QUIC_STATUS
 CxPlatSocketStartReceive(
     _In_ CXPLAT_SOCKET_PROC* SocketProc,
     _Out_opt_ ULONG* SyncIoResult,
-    _Out_opt_ uint16_t* SyncBytesReceived
+    _Out_opt_ uint16_t* SyncBytesReceived,
+    _Out_opt_ CXPLAT_DATAPATH_INTERNAL_RECV_CONTEXT** SyncRecvContext
     )
 {
     QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
     CXPLAT_DATAPATH* Datapath = SocketProc->Parent->Datapath;
     CXPLAT_DATAPATH_INTERNAL_RECV_CONTEXT* RecvContext;
     int Result;
     DWORD BytesRecv = 0;
+    WSABUF WsaBuf;
```

Comment:
Everything else is heap allocated, but this can be on the stack?

Reply:
Yes, the WSABUF is captured during the call to WsaRecv* but the other structures (WSAMSG, control buffer) need to be valid until the IO completes.